engine/access/rpc/backend/backend_transactions.go

```diff
@@ -609,42 +609,23 @@ func (b *backendTransactions) GetSystemTransactionResult(ctx context.Context, bl
 		return nil, rpc.ConvertStorageError(err)
 	}
 
-	req := &execproto.GetTransactionsByBlockIDRequest{
-		BlockId: blockID[:],
-	}
-	execNodes, err := executionNodesForBlockID(ctx, blockID, b.executionReceipts, b.state, b.log)
-	if err != nil {
-		if IsInsufficientExecutionReceipts(err) {
-			return nil, status.Errorf(codes.NotFound, err.Error())
+	if b.txResultCache != nil {
```

Comment:
I think caching this is generally a good idea, however this cache is specifically used for historic tx results and is generally not used. my preference is to not cache here, and return the result from `lookupTransactionResult`